### PR TITLE
feat: ci refactor

### PR DESCRIPTION
**Pull request by jhagestedt**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -127,31 +190,18 @@
...
-            <phase>test-compile</phase>
           </execution>
           <execution>
-            <id>jacoco-site</id>
-            <phase>verify</phase>
+            <id>report</id>
             <goals>
               <goal>report</goal>
             </goals>
+            <phase>verify</phase>
```
Comment (CCFenner) [inline pom.xml:204]: This is the [default](https://www.eclemma.org/jacoco/trunk/doc/report-mojo.html), can be omitted.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,9 +25,33 @@ jobs:
...
-    - run: |
-        mvn --batch-mode package
+    - run: mvn --batch-mode deploy
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    # will be uncommented as soon as sonar token is available
+    # - run: mvn --batch-mode verify sonar:sonar -Dsonar.login=${SONAR_TOKEN}
+    #   env:
+    #     SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+    #     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (CCFenner) [inline .github/workflows/ci.yml:35]:
```suggestion
```

File: pom.xml (hunk 1 above)
Comment (jhagestedt) [reply pom.xml:204]: right will change!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jhagestedt) [reply .github/workflows/ci.yml:35]: No otherwise sonarcloud will print error that github token is required.

File: pom.xml (hunk 1 above)
Comment (jhagestedt) [reply pom.xml]: https://github.com/corona-warn-app/cwa-verification-server/pull/57/commits/0eeb5bb84828afd77fed2e52ffe5d78d75ed3eef

File: pom.xml [hunk 3]
```diff
@@ -11,41 +11,76 @@
   <name>cwa-verification-server</name>
   <description>CWA verification server project.</description>
```
Comment (CCFenner) [inline pom.xml:13]: Add URLs to be picked up by Sonar:

```suggestion

   <url>https://www.coronawarn.app/</url>
   <ciManagement>
     <url>https://github.com/corona-warn-app/cwa-verification-server/actions?query=workflow%3Aci</url>
   </ciManagement>
   <issueManagement>
     <url>https://github.com/corona-warn-app/cwa-verification-server/issues</url>
   </issueManagement>
   <scm>
     <url>https://github.com/corona-warn-app/cwa-verification-server</url>
   </scm>
```

File: pom.xml [hunk 4]
```diff
@@ -11,41 +11,76 @@
...
     <springdoc.version>1.3.9</springdoc.version>
-    <jacoco.version>0.8.5</jacoco.version>
-    <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
-    <sonar.dynamicAnalysis>reuseReports</sonar.dynamicAnalysis>
-    <sonar.jacoco.reportPath>${project.basedir}/../target/jacoco.exec</sonar.jacoco.reportPath>
-    <sonar.language>java</sonar.language>
+    <!-- plugins -->
+    <plugin.checkstyle.version>3.1.1</plugin.checkstyle.version>
+    <plugin.sonar.version>3.6.1.1688</plugin.sonar.version>
+    <plugin.jacoco.version>0.8.4</plugin.jacoco.version>
```
Comment (CCFenner) [inline pom.xml:31]: Did downgrade happen by intend?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jhagestedt) [reply .github/workflows/ci.yml:35]: see blueprint
https://github.com/jhagestedt/blueprint/runs/703631738?check_suite_focus=true#step:6:502

File: pom.xml (hunk 4 above)
Comment (jhagestedt) [reply pom.xml:31]: whoops :D 0.8.5 is right - will change!

Comment (jhagestedt) [reply pom.xml]: https://github.com/corona-warn-app/cwa-verification-server/pull/57/commits/42663411402119049d852701219b474a4de99a1a

File: .github/workflows/ci.yml (hunk 2 above)
Comment (CCFenner) [reply .github/workflows/ci.yml:35]: Yes, sorry. I thought plan to extend the step before. If you have an additional step for `sonar:sonar` omit the `verify` to save execution time. `verify` is implicitly executed in the step before.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -23,9 +25,33 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/download-artifact@v1
+      with:
+        name: target
+        path: target
+    - run: sudo apt-get install --yes --no-install-recommends libxml-xpath-perl
+    - run: |
+        VERSION=$(xpath -q -e "/project/version/text()" pom.xml)
```
Comment (CCFenner) [inline .github/workflows/ci.yml:51]: Can be done using plain Maven:
```
mvn org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression=project.version -q -DforceStdout
```

Comment (jhagestedt) [reply .github/workflows/ci.yml:51]: yeah but on i think it takes longer - at least in bigger projects this was up to 90 seconds from my expirience

**Comment by jhagestedt** [comment]
@CCFenner are you admin of the sonarcloud.io organization?
Currently searching the creator to add our project...

**Pull request merged**